crates/bevy_ui/src/render/mod.rs

```diff
@@ -145,13 +145,47 @@ fn get_ui_graph(render_app: &mut App) -> RenderGraph {
     ui_graph
 }
 
+/// Texture size and subrect data
+#[derive(Clone, Copy, Debug)]
+pub enum ExtractedRect {
+    /// Render the entire texture with this size
+    Size(Vec2),
+    /// Select a subsection of the texture to render
+    Section { sub_rect: Rect, size: Vec2 },
+}
+
+impl ExtractedRect {
+    #[inline]
+    pub fn size(self) -> Vec2 {
+        match self {
```

Comment:
This adds a branch where there were none before. Potentially negatively impacting prepare stage perf.

Reply:
I don't think any new branches are added, they are just moved around. `prepare_uinodes` is very complicated though so it's hard to tell.

Reply:
I think it's actually an improvement overall, there is no size calculation when not using an atlas as we just pass it in directly.